src/main/java/com/epam/aidial/core/service/PublicationService.java

```diff
@@ -524,6 +532,26 @@ private void checkTargetResources(Publication publication) {
         }
     }
 
+    private void checkTargetResourcesExists(Publication publication) {
+        for (Publication.Resource resource : publication.getResources()) {
+            String url = resource.getTargetUrl();
+            ResourceDescription descriptor = ResourceDescription.fromPublicUrl(url);
+
+            if (!checkResource(descriptor)) {
+                throw new IllegalArgumentException("Target resource does not exist: " + descriptor.getUrl());
+            }
+        }
+    }
+
+    private void checkTargetResourceAccess(ProxyContext context, Publication publication) {
+        for (Publication.Resource resource : publication.getResources()) {
+            String url = resource.getTargetUrl();
+            if (!hasPublicAccess(context, ResourceDescription.fromPublicUrl(url))) {
```

Comment:
Please optimize it to check all resources at once.

Reply:
added